test/PreAMMBatcher-UnitTests.ts

```diff
@@ -0,0 +1,320 @@
+import { use, expect } from "chai";
```

Comment:
I think the standard JS/TS Mocha practice is to call them `ThingBeingTested.test.ts` or `ThingBeingTested.spec.ts`. Personally, I slightly prefer `*.test.ts` over `*.spec.ts`, but I'm not a fan of `*-UnitTest.ts` pattern, I find it unnecessarily verbose (and contains capitals!).

I would prefer renaming `PreAMMBatcher-e2e.ts -> PreAMMBatcher.e2e.ts` (`-` to `.`).